src/PostgREST/Response.hs

```diff
@@ -292,9 +295,32 @@ optionalRollback AppConfig{..} ApiRequest{iPreferences=Preferences{..}} resp = d
 -- | Add headers not already included to allow the user to override them instead of duplicating them
 addHeadersIfNotIncluded :: [HTTP.Header] -> [HTTP.Header] -> [HTTP.Header]
 addHeadersIfNotIncluded newHeaders initialHeaders =
-  filter (\(nk, _) -> isNothing $ find (\(ik, _) -> ik == nk) initialHeaders) newHeaders ++
+  filter (\(nk, nv) -> isNothing $ find (\(ik, iv) -> ik == nk && nv == iv) initialHeaders) newHeaders ++
   initialHeaders
 
+{-
+  Usage: Filters out multiple Preference-Applied Headers from the list and concatenate them into a single Preference-Applied header
+  Example:
+
+  Input: [(hContentType, "application/json")
+         , (hPreferenceApplied, "tx=commit")
+         , (hPreferenceApplied, "return=minimal")]
+  Output: [(hContentType, "application/json"
+          , (hPreferenceApplied, "tx=commit, return=minimal")]
+
+-}
+concatPrefAppsHeaders :: [HTTP.Header] -> [HTTP.Header]
+concatPrefAppsHeaders headers = otherHeaders ++ [(HTTP.hPreferenceApplied, combinedPrefApps)]
```

Comment:
This one looks like a good candidate for a `doctest` since it's a pure function. This way we can verify the input/output comment.

To enable it, you can add the `Response.hs` module here:

https://github.com/PostgREST/postgrest/blob/2d00d7d2488610cb14051400f91012de902473e0/test/doc/Main.hs#L8-L23

And then run it with `postgrest-test-doctests`.

Then you'd have to modify the comment so doctest can understand it. You can see the format on the other modules were `doctest` is used and also on https://github.com/sol/doctest/

Reply:
@steve-chavez I have added the doctest and I am receiving this strange error. What could be the problem here?
```haskell
-- |  Filters out multiple Preference-Applied Headers from the list and concatenate them into a single Preference-Applied header:
-- 
-- >>> :{ 
--  concatPrefAppsHeaders 
--     [("Content-Type", "application/json")
--     , ("Preference-Applied", "tx=commit")
--     , ("Preference-Applied", "return=minimal")]
-- :}
-- [("Preference-Applied", "application/json")
-- , ("Preference-Applied", "tx=commit, return=minimal")]
--

```

```
src/PostgREST/Version.hs:12:1: error:
    Could not find module ‘Paths_postgrest’
    Perhaps you meant Paths_doctest
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
src/PostgREST/Response.hs:319: failure in expression `:{ 
  concatPrefAppsHeaders 
     [("Content-Type", "application/json")
     , ("Preference-Applied", "tx=commit")
     , ("Preference-Applied", "return=minimal")]
:}'
expected: [("Preference-Applied", "application/json")
          , ("Preference-Applied", "tx=commit, return=minimal")]
 but got: 
          ^
          <interactive>:28:3: error:
              Variable not in scope: concatPrefAppsHeaders :: [(a0, b0)] -> t

```

Reply:
> I have added the doctest

@taimoorzaeem Could you commit the doctest here so I can see what's wrong?

Reply:
This is similar to the error we got in https://github.com/PostgREST/postgrest/pull/2740#issuecomment-1609961339. It looks like `Response.hs` imports `Version.hs` in a nested way through other modules (e.g. `PostgREST.Response.OpenAPI`).

Not sure what's the next step to solve this and keep the doctest. Maybe restructuring the files (a lot of work) or moving the function to another module that does not import `Version.hs`.

Reply:
@laurenceisla Good catch! Fixing that on https://github.com/PostgREST/postgrest/pull/2888. 

> moving the function to another module that does not import Version.hs.

Yeah, that makes sense as it makes the Response.hs module more pure. We should only import `Version.hs` on modules that do IO.

Reply:
@steve-chavez This fixed it. The `doctests` are working fine now. Thanks!